examples/sampling/graphbolt/node_classification.py

```diff
@@ -349,28 +362,35 @@ def parse_args():
         " identical with the number of layers in your model. Default: 10,10,10",
     )
     parser.add_argument(
-        "--device",
-        default="cpu",
-        choices=["cpu", "cuda"],
-        help="Train device: 'cpu' for CPU, 'cuda' for GPU.",
+        "--mode",
+        default="pinned-cuda",
+        choices=["cpu-cpu", "cpu-cuda", "pinned-cuda", "cuda-cuda"],
+        help="Dataset storage placement and Train device: 'cpu' for CPU and RAM,"
+        " 'pinned' for pinned memory in RAM, 'cuda' for GPU and GPU memory.",
     )
     return parser.parse_args()
 
 
 def main(args):
     if not torch.cuda.is_available():
-        args.device = "cpu"
-    print(f"Training in {args.device} mode.")
+        args.mode = "cpu-cpu"
+    print(f"Training in {args.mode} mode.")
+    args.storage_device, args.device = args.mode.split("-")
     args.device = torch.device(args.device)
 
     # Load and preprocess dataset.
     print("Loading data...")
     dataset = gb.BuiltinDataset("ogbn-products").load()
 
     graph = dataset.graph
-    # Currently the neighbor-sampling process can only be done on the CPU,
-    # therefore there is no need to copy the graph to the GPU.
     features = dataset.feature
+    # Move the dataset to the selected storage.
+    if args.storage_device == "pinned":
+        graph.pin_memory_()
```

Comment:
as we discussed, we'd better figure out a better way that requires less code in user side. maybe adding copy_to datapipe and pinning graph within `gb.DataLoader` is feasible.

Reply:
How does this file look with the updated way of moving the graph and features?

Reply:
The reason why there are 2 `copy_to`s in the dataloader is that I wanted the CPU sampling code path to be selectable by setting `--mode=cpu-cuda`. If this mode is deleted, then the last `copy_to` can be deleted safely. However, `pinned-cuda` will be preferable to `cpu-cuda` for all scenarios I can think of.

Reply:
I used `args.storage_device` instead of having separate `args.feature_device` and `args.graph_device` to not overcomplicate it.

Reply:
Minimized the changed code by adding a .to method that can also pin object.